### PR TITLE
[30]: Enable post to FB

### PR DESCRIPTION
**Pull request by JacobPernell**
# Description
Issue: https://github.com/mskalandunas/Robin/issues/30
Description: PR for the first half of this issue. Created a struct that holds test data to capture and bind JSON from a Postman POST request.

# Screenshots/GIFS
Screen Shot 2022-07-01 at 12 50 23 AM

# Reviewers
@mskalandunas

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -38,5 +37,21 @@ func main() {
 		return c.JSON(http.StatusOK, res)
 	})
 
+  
+  type Content struct {
+    Username string `json:"username"`
```
Comment (mskalandunas) [inline server/server.go:42]: We only need the content body in this struct.  A username is not part of the content and the content id won't exist until after the POST request has been executed, as a POST is going to create a new resource.

File: server/server.go [hunk 2]
```diff
@@ -38,5 +37,21 @@ func main() {
...
+    ContentID string `json:"id"`
+    ContentBody string `json:"content_body"`
+  }
+
+  e.POST("/postTest", func(c echo.Context) (err error) {
```
Comment (mskalandunas) [inline server/server.go:47]: We shouldn't call this postTest -- this is meant to be our actual production content creation endpoint.

File: server/server.go [hunk 3]
```diff
@@ -38,5 +37,21 @@ func main() {
 		return c.JSON(http.StatusOK, res)
 	})
 
+  
+  type Content struct {
+    Username string `json:"username"`
+    ContentID string `json:"id"`
+    ContentBody string `json:"content_body"`
+  }
```
Comment (mskalandunas) [inline server/server.go:45]: This struct should be placed outside of the context of this function similar to the other types.

File: server/server.go [hunk 4]
```diff
@@ -38,5 +37,21 @@ func main() {
...
+
+  e.POST("/postTest", func(c echo.Context) (err error) {
+    content := new(Content)
+    if err = c.Bind(content); err != nil {
+      return
+    }
```
Comment (mskalandunas) [inline server/server.go:51]: Do we want to add something a little more helpful for error handling?

**Comment by JacobPernell** [comment]
Made fixes based on these comments, and I think the struct is what we need it to be for now, just handling some basic text content.

Screen Shot 2022-07-01 at 2 17 51 PM

@mskalandunas

File: server/server.go [hunk 5]
```diff
@@ -11,15 +12,14 @@ type ContentResponseMeta struct {
 }
 
 type Content struct {
-	Body string `json:"body"`
+	Body string `json:"content_body"`
```
Comment (JacobPernell) [inline server/server.go]: Just changed `Body` to `ContentBody`

**Review by mskalandunas — APPROVED** [review]

**Pull request merged**